src/generate-catalyst-data.ts

```diff
@@ -156,3 +164,12 @@ function findDummyItemWithSpecificName(DummyItemName: string) {
       i.itemCategoryHashes?.includes(ItemCategoryHashes.Dummies),
   );
 }
+
+function findAutoAppliedCatalystForCatalystPCH(catalystPCH: number) {
+  const socket = allsockets.find((sockets) =>
+    sockets.plugWhitelist?.find((plug) => plug.categoryHash === catalystPCH),
+  );
+
+  return socket?.plugWhitelist?.find((plug) => plug.categoryHash === catalystPCH)
+    ?.reinitializationPossiblePlugHashes?.[0];
```

Comment:
Is there a way to simplify this to avoid the repeated call? Would the following be more or less performant?
```
function findAutoAppliedCatalystForCatalystPCH(catalystPCH: number) {
  const plug = allsockets
    .flatMap((sockets) => sockets.plugWhitelist || [])
    .find((plug) => plug.categoryHash === catalystPCH);

  return plug?.reinitializationPossiblePlugHashes?.[0];
}
```
(I think there's also a slight difference in meaning between "from all sockets, find first socket with a matching plugWhitelist PCH entry and pull that plugWhitelist's reinit field" vs "from all plugWhitelist entries, find the first one with a matching PCH and pull its reinit field" but that shouldn't affect anything)

Reply:
I went ahead and made the change, since it seemed to perform about the same and is way more readable.